### PR TITLE
rbd: Document new RBD feature flags and version support

### PR DESCRIPTION
**Pull request by ProgVal**
Sources:

layering:

* https://github.com/ceph/ceph/commit/f87c441f6891e5e72d5e712d9a19e45a1b9a97eb
* krbd: https://github.com/torvalds/linux/commit/770eba6e295fd36e43881176ee0644b9cc2803f1

striping:

* https://github.com/ceph/ceph/commit/2e403ef2c98bb64220060c06b435fb1ffe08294e
* krbd: https://github.com/torvalds/linux/commit/770eba6e295fd36e43881176ee0644b9cc2803f1

exclusive-lock:

* https://github.com/ceph/ceph/commit/fa96bb4db47c500947efcab51892f45a04b03d30
* krbd: https://github.com/torvalds/linux/commit/ed95b21a4b0a71ef89306cdeb427d53cc9cb343f

object-map:

* https://github.com/ceph/ceph/commit/7e751ce9125cbeb04b9551afff4b91c6255c03f9

fast-diff:

* https://github.com/ceph/ceph/commit/16ad44cebcc5d2464be38b95f16e0d63d2b29ed1

deep-flatten:

* https://github.com/ceph/ceph/commit/b556d3173ae4483e681bb21658a7d484a2627386

journaling:

* https://github.com/ceph/ceph/commit/750771c6ff07f6fcf0235172ec68ff4d907c621d
* not a default: https://github.com/ceph/ceph/blob/1a14963/src/test/cli/rbd/help.t#L229

data-pool:

* https://github.com/ceph/ceph/commit/c5f22900474e0eed88910341de57e17e1cc154f0
* description: https://tracker.ceph.com/issues/17409
* krbd: https://github.com/torvalds/linux/commit/7e97332ea9caad3b7c6d86bc3b982e17eda2f736

operations:

* https://github.com/ceph/ceph/commit/209f6fc5a44e2d068ce0dd8b3e9a25ee700ad0e0
* description: https://github.com/torvalds/linux/commit/e573427a440fd67d3f522357d7ac901d59281948
* krbd: https://github.com/torvalds/linux/commit/e573427a440fd67d3f522357d7ac901d59281948

migrating:

* https://github.com/ceph/ceph/commit/2f7c4ae3fc7ec7a2bfe99e116bf5e4f71f24c6a1

Signed-off-by: Valentin Lorentz <progval+git@progval.net>

### CODE REVIEW

**Comment by ProgVal** [comment]
There's an issue with Jenkins:

> stderr: fatal: unable to access 'https://github.com/ceph/ceph.git/': Could not resolve host: github.com

**Comment by LenzGr** [comment]
retest this please

**Comment by LenzGr** [comment]
Whoops, clicked the wrong button by accident. Sorry about that.

**Comment by LenzGr** [comment]
jenkins render docs

File: doc/rbd/rbd-config-ref.rst [hunk 1]
```diff
@@ -146,45 +146,90 @@ RBD supports advanced features which can be specified via the command line when
 :Description: Layering enables you to use cloning.
 :Internal value: 1
 :CLI value: layering
+:Added in: 0.70
+:KRBD support: v3.10-rc1
```
Comment (dillaman) [inline doc/rbd/rbd-config-ref.rst]: Nit: might as well just say ```since v3.10```

File: doc/rbd/rbd-config-ref.rst [hunk 2]
```diff
@@ -146,45 +146,90 @@ RBD supports advanced features which can be specified via the command line when
 :Description: Layering enables you to use cloning.
 :Internal value: 1
 :CLI value: layering
+:Added in: 0.70
```
Comment (dillaman) [inline doc/rbd/rbd-config-ref.rst]: Nit: some librbd versions state ```vX.Y.Z``` vs ```X.Y.Z``` -- perhaps standardize on the ```vX.Y.Z``` or just use the release name (i.e. ```Jewel```)

File: doc/rbd/rbd-config-ref.rst [hunk 3]
```diff
@@ -146,45 +146,90 @@ RBD supports advanced features which can be specified via the command line when
...
+:Description: Used to restrict older clients from performing certain maintenance operations against an image (e.g. clone, snap create).
+:Internal value: 256
+:Added in: v13.0.2 (Mimic)
+:KRBD support: since v4.16
+
+
+``Migrating``
+
+:Internal value: 512
+:Added in: v14.0.1
```
Comment (dillaman) [inline doc/rbd/rbd-config-ref.rst]: Nit: ```v14.0.1 (Nautilus)```

Comment (trociny) [reply doc/rbd/rbd-config-ref.rst]: And for "Description" you could have something like this:

Used to restrict older clients from opening an image when it is in migration state.

Comment (ProgVal) [reply doc/rbd/rbd-config-ref.rst]: Thanks!

**Comment by dillaman** [comment]
@ProgVal lgtm -- can you squash your commits to a single commit w/ the "doc" prefix?

**Comment by ProgVal** [comment]
done

**Pull request merged**